openraft/src/membership/membership.rs

```diff
@@ -19,6 +19,9 @@ use crate::NodeId;
 /// every config.
 #[derive(Clone, Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Membership {
+    /// learners set
+    learners: BTreeSet<NodeId>,
+
```

Comment:
Is it allowed to have a `node_id` existing in both `learners` and `configs`?

It's better to constrain it by either:
- `leaners` has to include all node ids that are in `configs,`
- or `learners` has to include none of the node in `configs.`

With this constraint, the leader will not need to check for duplicated streams when spawning replication tasks.